### PR TITLE
Improve Lambda Generation and optimize preset trade-offs

### LINKED ISSUE #1449: Align presets to svt-02-temp
Current trade-offs compared to the svt-02-temp branch can be improved.  Should align lambda generation and preset features to improve trade-offs.

### PR DESCRIPTION
**Pull request by PhoenixWorthVCD**
# Description

- Use temporal layer information to improve lambda generation.
- Shift features to optimize preset trade-offs.
-- Cleanup: Remove unused signals

# Issue

Fixes #1449 

# Author(s)

@PhoenixWorthVCD 

# Performance impact

- [x] quality
- [ ] memory
- [x] speed
- [ ] 8 bit
- [ ] 10 bit
- [ ] N/A

# Test set
- [ ] obj-1-fast can be found [here](https://media.xiph.org/video/derf/objective-1-fast.tar.gz)
- [x] Google Lowres
- [ ] N/A

| Preset(s)| BD-Rate  Deviation| Speed Deviation| 
| -- | -- | --
|M0 | -0.33% | No change|
|M1 | -0.17% | 6.5%|
|M2 | -0.45% | No change|
|M3 | -0.41% | No change|
|M4 | -0.73% | No change|
|M5 | -0.81% | -3%|
|M6 | -0.54% |  -3%|
|M7 | -1.01% | -4%|

# Merge method
- [x] Allow the maintainer to squash and merge when PR is ready to create a 1-commit to the master branch. The maintainer will be able to fix typos / combine commit messages to create a more readable 1-commit message or use whatever is stated in the 'Description' section
- [ ] I will clean up my commits and the maintainer shall use 'rebase and merge' to the master branch

### CODE REVIEW

File: Source/Lib/Encoder/Codec/EbModeDecisionProcess.c [hunk 1]
```diff
@@ -655,20 +671,38 @@ void av1_lambda_assign_md(
...
         *full_lambda = av1lambda_mode_decision10_bit_sse[qp_index];
+#endif
         *fast_lambda = av1lambda_mode_decision10_bit_sad[qp_index];
         if (multiply_lambda) {
             *full_lambda *= 16;
             *fast_lambda *= 4;
         }
     } else if (bit_depth == 12) {
+#if USE_GF_UPDATE_FOR_LAMBDA
+        *full_lambda = (uint32_t)compute_rdmult_sse(pcs_ptr, (uint8_t)qp_index, bit_depth);
```
Comment (hassount) [inline Source/Lib/Encoder/Codec/EbModeDecisionProcess.c:702]: we have a case where bit-depth == 12 ?

File: Source/Lib/Encoder/Codec/EbModeDecisionProcess.c [hunk 2]
```diff
@@ -655,20 +671,38 @@ void av1_lambda_assign_md(
...
+#else
         *full_lambda = av1lambda_mode_decision12_bit_sse[qp_index];
+#endif
         *fast_lambda = av1lambda_mode_decision12_bit_sad[qp_index];
     } else {
         assert(bit_depth >= 8);
```
Comment (hassount) [inline Source/Lib/Encoder/Codec/EbModeDecisionProcess.c:708]: shouldn't the assert here be for bd > 12 ?

File: Source/Lib/Encoder/Codec/EbModeDecisionProcess.c (hunk 1 above)
Comment (PhoenixWorthVCD) [reply Source/Lib/Encoder/Codec/EbModeDecisionProcess.c:702]: There was a case for bit_depth ==12 already, so I updated the lambda generation to use the new function.

File: Source/Lib/Encoder/Codec/EbModeDecisionProcess.c (hunk 2 above)
Comment (PhoenixWorthVCD) [reply Source/Lib/Encoder/Codec/EbModeDecisionProcess.c:708]: There are two asserts (not added by this PR; they are already in the code) that cover the range of allowable bit depths:

```
        assert(bit_depth >= 8);
        assert(bit_depth <= 12);
```

**Comment by EricLamVCD** [comment]
Tested obj-1 8bit and 10bit clips: 
  | Speed Deviation  | BD-Rate   Deviation | Memory   Deviation
-- | -- | -- | --
M0 - 8bit | -0.80% | 0.00% | -0.09%
M2 - 8bit | 1.70% | -0.20% | -0.31%
M3 - 8bit | 1.74% | -0.43% | 0.15%
M4 - 8bit | -10.67% | -1.18% | 1.11%
M5 - 8bit | 0.06% | -1.15% | 0.52%
M6 - 8bit | -7.11% | -1.23% | 0.00%
M7 - 8bit | -5.71% | -3.52% | -0.12%
M0 - 10bit | -0.81% | -0.03% | -0.01%
M1 - 10bit | 17.25% | 0.51% | 0.04%
M2 - 10bit | 4.44% | -0.30% | 0.05%
M3 - 10bit | 1.64% | -0.44% | 0.04%
M4 - 10bit | -9.82% | -1.24% | -0.29%
M5 - 10bit | 0.40% | -1.15% | -0.49%
M6 - 10bit | -9.63% | -1.60% | 0.91%
M7 - 10bit | -6.19% | -3.34% | 3.13%

**Pull request merged**